lgc/include/lgc/state/TargetInfo.h

```diff
@@ -138,6 +138,28 @@ class TargetInfo {
   // Set TargetInfo. Returns false if the GPU name is not found or not supported.
   bool setTargetInfo(llvm::StringRef gpuName);
 
+  // GFX IP checkers (use invalid value to denote this info is a don't-care value)
+  bool isGfx(unsigned major, unsigned minor = InvalidValue, unsigned stepping = InvalidValue) const { // GFX(IP)
+    if (minor != InvalidValue && stepping != InvalidValue) // Check (major, minor, stepping)
+      return std::tie(m_gfxIp.major, m_gfxIp.major, m_gfxIp.stepping) == std::tie(major, minor, stepping);
+    if (minor != InvalidValue) // Check (major, minor)
+      return std::tie(m_gfxIp.major, m_gfxIp.major) == std::tie(major, minor);
+    return m_gfxIp.major == major; // Just check major
+  }
+  bool isGfxAbove(unsigned major, unsigned minor = InvalidValue, unsigned stepping = InvalidValue) const { // > GFX(IP)
```

Comment:
nit: maybe `isGfxNewerThan(...)`, `isGfxOlderThan(...)`, `isGfxNewerOrEqualThan(...)`

Reply:
The terminologies "above", "below", "plus" are defined by HW in their register definitions. I just borrow the items :)